### ISSUE TITLE
Update readme with proposal

### ISSUE THREAD

**Issue by sbabicki** [UBC-MDS/kmeaningful#7]
#3 

Outline the package you would like to build in the README.md file. (This can be identical for both projects at this point in the project). In particular, your README.md should contain:

- [x] a summary paragraph that describes the project at a high level

- [x] a bulleted list of the functions (and datasets if applicable) that will be included in the package (this should be a 1-2 sentence description for each function/dataset)

- [x] a paragraph describing where your packages fit into the Python ecosystem (are there any other Python packages that have the same/similar functionality? Provide links to any that do. If none exist, then clearly state this as well).

**Comment by sbabicki** [comment]
@mikelynch416 to review

**Comment by sbabicki** [comment]
@trevorki : a paragraph describing where your packages fit into the Python ecosystem (are there any other Python packages that have the same/similar functionality? Provide links to any that do. If none exist, then clearly state this as well).
@sbabicki : a bulleted list of the functions (and datasets if applicable) that will be included in the package (this should be a 1-2 sentence description for each function/dataset)
@HazelJJJ : a summary paragraph that describes the project at a high level, compile list, and update README

**Comment by sbabicki** [comment]
# Features 
- `fit_and_predict(X, k)` - K-Means implementation, calls `fit(X, k)` and `predict(X)`
- `preprocess(X)` - Automatic dataset preprocessing
- `find_elbow(X)` -  Automatic hyperparameter tuning
- `show_fit(X, k)` - Visualize clusters according to PCA

**Comment by trevorki** [comment]
There already exist several packages that implement k-means clustering in python. Most notably there are the Scikit-learn [`sklearn.cluster.KMeans`](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html) and SciPy [`scipy.cluster.vq.kmeans`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.cluster.vq.kmeans.html#scipy.cluster.vq.kmeans) implementations. We are not trying to break new ground with `kmeaningful`, but rather to build a simple and lightweight implementation from scratch.

**Comment by HazelJJJ** [comment]
Have you ever encountered a dataset that seems to have different patterns in it? Have you ever considered to group similar things together in a dataset and to make prediction for a new sample? We created `kmeaningful` to help with it. `kmeaningful` is a Python package that use algorithm to assign labels to clusters and make prediction to new data points. It also contains functions to help with data preprocessing, hyperparameter tuning and visualizing clusters.

**Comment by mikelynch416** [comment]
Does this look like a reasonable incorporation of everyone's parts into the readme?

# kmeaningful 

![](https://github.com/UBC-MDS/kmeaningful/workflows/build/badge.svg) [![codecov](https://codecov.io/gh/UBC-MDS/kmeaningful/branch/main/graph/badge.svg)](https://codecov.io/gh/UBC-MDS/kmeaningful) ![Release](https://github.com/UBC-MDS/kmeaningful/workflows/Release/badge.svg) [![Documentation Status](https://readthedocs.org/projects/kmeaningful/badge/?version=latest)](https://kmeaningful.readthedocs.io/en/latest/?badge=latest)

Have you ever encountered an unlabeled dataset that seems to have different patterns in it? Have you ever considered grouping similar things together in a dataset and predicting which group a new sample should go in? We created `kmeaningful` to help with these problems. `kmeaningful` is a Python package that uses the K-means algorithm to assign clusters to unlabeled data and to make prediction for the cluster assignments of new data points. It also contains functions to help with data preprocessing, hyperparameter tuning and visualizing clusters.

## Kmeaningful's Place in the Python Ecosystem

There already exist several packages that implement k-means clustering in python. Most notably there are the Scikit-learn [`sklearn.cluster.KMeans`](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html) and SciPy [`scipy.cluster.vq.kmeans`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.cluster.vq.kmeans.html#scipy.cluster.vq.kmeans) implementations. We are not trying to break new ground with `kmeaningful`, but rather to build a simple and lightweight implementation from scratch.

## Installation

```bash
$ pip install -i https://test.pypi.org/simple/kmeaningful
```

## Features

- `fit_and_predict(X, k)` - K-Means implementation, calls `fit(X, k)` and `predict(X, centres)`
- `preprocess(X)` - Automatic dataset preprocessing
- `find_elbow(X)` -  Automatic hyperparameter tuning
- `show_fit(X, k)` - Visualize clusters according to PCA

## Dependencies

- TODO

## Usage

- TODO

## Documentation

The official documentation is hosted on Read the Docs: https://kmeaningful.readthedocs.io/en/latest/

## Contributors
This project was created by DSCI 524 Group 16: 
- Yihong (Hazel) Jiang
- Mike Lynch
- Trevor Kinsey
- Sasha Babicki

We welcome and recognize all contributions. You can see a list of current contributors in the [contributors tab](https://github.com/UBC-MDS/kmeaningful/graphs/contributors).

### Credits

This package was created with Cookiecutter and the UBC-MDS/cookiecutter-ubc-mds project template, modified from the [pyOpenSci/cookiecutter-pyopensci](https://github.com/pyOpenSci/cookiecutter-pyopensci) project template and the [audreyr/cookiecutter-pypackage](https://github.com/audreyr/cookiecutter-pypackage).

**Comment by sbabicki** [comment]
Yeah, I like the header. I'll incorporate this into the README right now and send a pull request.
